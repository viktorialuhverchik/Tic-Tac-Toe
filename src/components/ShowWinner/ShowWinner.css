.show-winner {
    display: flex;
    justify-content: center;
    align-items: center;
}

.message {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    max-width: 310px;
    height: 100px;
}

.message-title {
    color: #69524f;
}

.button-again {
    margin-bottom:0.5em;
    width: 150px;
    height: 45px;
    background-color: #b9a195;
    border: 1px solid #b99e90;
    border-radius: 0.5em;
    outline: none;
    cursor: pointer;
    box-shadow: 2px 2px 3px #867e7d;
}
