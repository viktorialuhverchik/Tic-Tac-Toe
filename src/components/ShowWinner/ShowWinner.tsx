import React, { FC } from 'react';
import { useDispatch } from 'react-redux';
import { initGame } from '../../redux/actions/actions';
import { PropsShowWinner } from '../../types';

import './ShowWinner.css';

const ShowWinner: FC<PropsShowWinner> = ({  winner }) => {

    const dispatch: any = useDispatch();

    return (
        <div className="show-winner">
            <div className="message">
                <h1 className="message-title">{winner}</h1>
                <button className="button-again" data-testid="again" onClick={() => dispatch(initGame())}>
                    <span className="button-title">Play again</span>
                </button>
            </div>
        </div>
    );

};

export default ShowWinner;
