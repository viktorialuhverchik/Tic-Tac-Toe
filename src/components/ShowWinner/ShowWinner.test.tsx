import React from 'react';
import ShowWinner from './ShowWinner';
import { renderWithRedux } from '../../index.test';

let winner: string = "";

describe('ShowWinner component',() => {
    it('should render ShowWinner component', () => {
        const wrapper = renderWithRedux(<ShowWinner winner={winner} />);
        expect(wrapper).toMatchSnapshot();
    });
});
