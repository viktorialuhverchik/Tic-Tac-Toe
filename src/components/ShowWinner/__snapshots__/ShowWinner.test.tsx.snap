// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowWinner component should render ShowWinner component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="show-winner"
      >
        <div
          class="message"
        >
          <h1
            class="message-title"
          />
          <button
            class="button-again"
          >
            <span
              class="button-title"
            >
              Play again
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="show-winner"
    >
      <div
        class="message"
      >
        <h1
          class="message-title"
        />
        <button
          class="button-again"
        >
          <span
            class="button-title"
          >
            Play again
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
